src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkImageBrowser.tsx

```diff
@@ -33,73 +39,104 @@ export const ArtworkImageBrowser = (props: ArtworkBrowserProps) => {
   )
 }
 
+/*
+ * Returns null if no window for SSR
+ */
+const isClient = typeof window !== "undefined"
+const Flickity = isClient ? require("react-flickity-component") : () => null
+
 export class LargeArtworkImageBrowser extends React.Component<
-  ArtworkBrowserProps
+  ArtworkBrowserProps,
+  ArtworkBrowserState
 > {
-  slider: Slider
+  state = { mounted: false }
 
-  setSliderRef = slider => {
-    this.slider = slider
-    if (this.props.sliderRef) {
-      this.props.sliderRef(slider)
+  flickity = null
+
+  options = {
+    prevNextButtons: false,
+    wrapAround: true,
+    pageDots: true,
+    cellAlign: "left",
+    draggable: false,
+    lazyLoad: true,
+  } as FlickityOptions
+
+  componentDidMount() {
+    this.setState({
+      mounted: true,
+    })
+    const { setFlickity } = this.props
+    if (this.flickity && setFlickity) {
+      setFlickity(this.flickity)
     }
   }
 
-  get settings(): Settings {
-    return {
-      arrows: false,
-      customPaging: () => <PageIndicator />,
-      dots: true,
-      infinite: false,
-      lazyLoad: "ondemand",
-      speed: 0,
-      swipe: false,
-      slidesToShow: 1,
-      slidesToScroll: 1,
-    }
+  renderSlide = (image, hasMultipleImages) => {
+    return (
+      <Flex
+        flexDirection="column"
+        justifyContent="center"
+        width="100%"
+        px={hasMultipleImages ? [2, 2, 0] : 0}
+        key={isClient + image.id}
+      >
+        <Lightbox
+          deepZoom={image.deepZoom}
+          enabled={image.is_zoomable}
+          isDefault={image.is_default}
+        >
+          <DesktopImage src={image.uri} width="100%" />
+        </Lightbox>
+      </Flex>
+    )
+  }
+
+  renderCarousel = hasMultipleImages => {
+    return (
+      <Flickity options={this.options} flickityRef={c => (this.flickity = c)}>
+        {this.props.images.map(image => {
+          return this.renderSlide(image, hasMultipleImages)
+        })}
+      </Flickity>
+    )
+  }
+
+  renderServerCarousel = hasMultipleImages => {
+    return (
+      <Flex>
+        {this.props.images.map((image, index) => {
+          const isFirstImage =
+            index === 0 ? this.renderSlide(image, hasMultipleImages) : null
+          return isFirstImage
+        })}
+      </Flex>
+    )
```

Comment:
Same here -- we shouldn't have to scaffold components in order to use Flickity. Behind the scenes this kind of setup can happen so that at the consumer level its usable without needing explicit setup instructions.

Reply:
I'm not sure this is possible with all of the rehydration and ref passing but I can timebox looking into it.

Reply:
If needed we can work through some designs to help keep things as simple as we can. 

One strategy I often employ is to first write out exactly what I would _like_ to use, like the ideal component / api as it would eventually be used, and then after that feels right I then work backwards and retrofit the underlying code to make everything work according to the ideal. This approach sometimes takes a bit more upfront planning (not necessarily so, tho) but sometimes working backwards helps one work forwards in a better way.

Reply:
(I can also take a look at the new Flickity scroller to see if there's anything we can do to make sure the implementation details are contained a bit more.)